servicetalk-grpc-netty/src/main/java/io/servicetalk/grpc/netty/GrpcToHttpLifecycleObserverBridge.java

```diff
@@ -0,0 +1,178 @@
+import static java.util.Objects.requireNonNull;
+
+final class GrpcToHttpLifecycleObserverBridge implements HttpLifecycleObserver {
+
+    private final GrpcLifecycleObserver observer;
+
+    GrpcToHttpLifecycleObserverBridge(final GrpcLifecycleObserver observer) {
+        this.observer = requireNonNull(observer);
+    }
+
+    @Override
+    public HttpExchangeObserver onNewExchange() {
+        return new GrpcToHttpExchangeObserverBridge(observer.onNewExchange());
+    }
+
+    private static final class GrpcToHttpExchangeObserverBridge implements HttpExchangeObserver {
+
+        private final GrpcExchangeObserver observer;
+
+        GrpcToHttpExchangeObserverBridge(final GrpcExchangeObserver observer) {
+            this.observer = observer;
+        }
+
+        @Override
+        public void onConnectionSelected(final ConnectionInfo info) {
+            observer.onConnectionSelected(info);
+        }
+
+        @Override
+        public HttpRequestObserver onRequest(final HttpRequestMetaData requestMetaData) {
+            return new GrpcToHttpRequestObserver(observer.onRequest(requestMetaData));
+        }
+
+        @Override
+        public HttpResponseObserver onResponse(final HttpResponseMetaData responseMetaData) {
+            return new GrpcToHttpResponseObserver(observer.onResponse(responseMetaData), responseMetaData);
+        }
+
+        @Override
+        public void onResponseError(final Throwable cause) {
+            observer.onResponseError(cause);
+        }
+
+        @Override
+        public void onResponseCancel() {
+            observer.onResponseCancel();
+        }
+
+        @Override
+        public void onExchangeFinally() {
+            observer.onExchangeFinally();
+        }
+    }
+
+    private static final class GrpcToHttpRequestObserver implements HttpRequestObserver {
+
+        private final GrpcRequestObserver observer;
+
+        GrpcToHttpRequestObserver(final GrpcRequestObserver observer) {
+            this.observer = observer;
+        }
+
+        @Override
+        public void onRequestData(final Buffer data) {
+            observer.onRequestData(data);
+        }
+
+        @Override
+        public void onRequestTrailers(final HttpHeaders trailers) {
+            observer.onRequestTrailers(trailers);
+        }
+
+        @Override
+        public void onRequestComplete() {
+            observer.onRequestComplete();
+        }
+
+        @Override
+        public void onRequestError(final Throwable cause) {
+            observer.onRequestError(cause);
+        }
+
+        @Override
+        public void onRequestCancel() {
+            observer.onRequestCancel();
+        }
+    }
+
+    private static final class GrpcToHttpResponseObserver implements HttpResponseObserver {
+
+        private static final CharSequence GRPC_STATUS_CODE_TRAILER = newAsciiString("grpc-status");
+        private static final CharSequence GRPC_STATUS_MESSAGE_TRAILER = newAsciiString("grpc-message");
```

Comment:
I was not able to reuse `GrpcUtils` because it's a pkg-private class in `grpc-api`. I also could not find a way to extract it into a separate internal module because of "chicken & egg" problem: it depends on `GrpcStatus` from `grpc-api` and `grpc-api` depends on these constants.